eZ/Publish/API/Repository/Tests/SearchService/RemoteIdIndexingTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Tests\SearchService;
+
+use eZ\Publish\API\Repository\Tests\BaseTest;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+final class RemoteIdIndexingTest extends BaseTest
+{
+    private const REMOTE_IDS = ['external:10', 'external_10'];
+
+    /**
+     * @dataProvider providerForTestIndexingRemoteId
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testIndexingRemoteId(Criterion $criterion): void
+    {
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+
+        // always prepare all the data to make sure only one is matched
+        $expectedContentIds = [];
```

Comment:
q: I might be rusty on phpunit, but who is destroying the created contents between each iteration of the dataProvider ? is the whole eZ db reset ?

Reply:
> q: I might be rusty on phpunit, but who is destroying the created contents between each iteration of the dataProvider ? is the whole eZ db reset ?

Each data provider causes PHPUnit to go through setUp() -> test -> tearDown() process.

Database is initialized in `\eZ\Publish\API\Repository\Tests\SetupFactory\Legacy::getRepository` (schema is not touched, if initialized, but data is dropped and re-created).

See `\eZ\Publish\API\Repository\Tests\BaseTest::getRepository` and `\eZ\Publish\API\Repository\Tests\BaseTest::tearDown`.

I could go for trying to preserve data for this test case, but this was a bit unstable in the past, especially when it comes to in-memory SQLite tests.
Most reasonable approach IMHO would be to create data in data provider, but it's a bit tricky - all data providers are executed before `setUp` test, so while it might be working now, when someone adds something, it might crash. It behaves differently when executed along with all the test suites and differently when executed standalone.

Test setup is one of the things that I need to improve (work in progress).